Create USGS_RCMAP_RELEASES_V5_COMPONENTS

ajgriebel opened the pull request:
Descriptions of the 9 RCMAP V5 Cover Components. New component area added for V5 - Tree.

schwehr commented:
Thanks for your pull request to the Earth Engine public data catalog!  

I will work with you to get this contribution to be ready for submission. I will be making an initial review comment that fixes a parsing problem.  Once you have fixed that and push the change to your branch, the checker should run and give the a bunch of issues (I tried the fix in a local copy on my machine).  I will  add review comments on how to approach all of them.  There will also be additional comments to discuss that the checker does not yet cover or cannot check. It will take a few iterations to get everything sorted.

schwehr commented on catalog/USGS/NLCD_RELEASES/USGS_RCMAP_RELEASES_V5_COMPONENTS.jsonnet:
```diff
@@ -0,0 +1,358 @@
+      maximum: 100.0,
+      'gee:estimated_range': false,
+    },
+    },
```
This line is the cause of the jsonnet file not being parsed.  Please remove it.

The build and check process showed this error:

```
ERROR: /home/runner/work/earthengine-catalog/earthengine-catalog/catalog/BUILD:6:17: Compiling Jsonnet to JSON for jsonnet_to_json_USGS_NLCD_RELEASES_USGS_RCMAP_RELEASES_V5_COMPONENTS.jsonnet failed: (Exit 1): bash failed: error executing command /bin/bash -c ... (remaining 1 argument skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
catalog/USGS/NLCD_RELEASES/USGS_RCMAP_RELEASES_V5_COMPONENTS.jsonnet:357:4-5 Did not expect: ","

  },
```

ajgriebel replied:
removed line 351-had an extra "}," in there

simonff commented on catalog/USGS/NLCD_RELEASES/USGS_RCMAP_RELEASES_V5_COMPONENTS.jsonnet:
```diff
@@ -0,0 +1,358 @@
+  ],
+  id: id,
+  title: 'Rangeland Condition Monitoring Assessment and Projection (RCMAP) Fractional Component Time-Series Across the Western U.S. 1985-2021',
+  version: 'v5',
```
this is a replacement for https://developers.google.com/earth-engine/datasets/catalog/USGS_NLCD_RELEASES_2019_REL_RCMAP_V4_COVER , right? Should the title be the same?

Also, could you please add versioning (successor / predecessor / latest) links to both v4 and v5 .  See the two CAS PML files here for an example: https://github.com/google/earthengine-catalog/tree/main/catalog/CAS

ajgriebel replied:
I did base it off of that file yes. Should the new one be renamed to: USGS_NLCD_RELEASES_2021_REL_RCMAP_V5_COVER?
2021 is the latest year available with this release, and it is V5. 
I'm not sure what you mean by versioning links. Do you mean in the Title? They do have V4 and V5 in the Titles, and it is identified on line 26.

ajgriebel replied:
Changed to USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_COVER

simonff commented on catalog/USGS/NLCD_RELEASES/USGS_RCMAP_RELEASES_V5_COMPONENTS.jsonnet:
```diff
@@ -0,0 +1,358 @@
+  links: ee.standardLinks(subdir, id) + [
+  ],
+  keywords: [
+    'shrublands',
```
keywords must be sorted and not include spaces

ajgriebel replied:
Updated to alphabetical order. Not sure on the spaces. please check that again and let me know more specifics.

simonff commented on catalog/USGS/NLCD_RELEASES/USGS_RCMAP_RELEASES_V5_COMPONENTS.jsonnet:
```diff
@@ -0,0 +1,358 @@
+local id = 'USGS/NLCD_RELEASES/USGS_RCMAP_RELEASES_V5_COMPONENTS';
```
please add an entry for this file to catalog.json in the same directory

ajgriebel replied:
I believe I did this. Added a line (line 24) in the catalog.jsonnet
ee.link.child_collection('USGS_NLCD_RELEASES_2021_REL_RCMAP_V5_COVER', base_url),

simonff commented on catalog/USGS/NLCD_RELEASES/USGS_RCMAP_RELEASES_V5_COMPONENTS.jsonnet:
```diff
@@ -0,0 +1,358 @@
+    Rigge, M.B., Bunde, B., Postma, K., Shi, H., 2022, Rangeland Condition Monitoring Assessment and Projection (RCMAP) Fractional Component Time-Series Across the Western U.S. 1985-2021: U.S. Geological Survey data release, https://doi.org/10.5066/P9ODAZHC.
+    [doi:10.5066/P9ODAZHC](https://doi.org/10.5066/P9ODAZHC)
+  |||,
+  'gee:user_uploaded': true,
```
please also add
  'gee:skip_indexing': true,
  (this is a temporary thing our system needs right now)

ajgriebel replied:
Added this line as line 108

ajgriebel commented on catalog/USGS/NLCD_RELEASES/USGS_RCMAP_RELEASES_V5_COMPONENTS.jsonnet:
```diff
@@ -0,0 +1,358 @@
+local self_url = catalog_subdir_url + base_filename;
+
+{
+  stac_version: ee_const.stac_version,
```
I believe I added this-line 17 (check that and let me know if it needs to be fixed)

ajgriebel commented:
@schwehr @simonff  Please update these 3 areas in the **RCMAP.libsonnet** (Thank you!) 
**Description**

The RCMAP (Rangeland Condition Monitoring Assessment and Projection) dataset quantifies the percent cover of rangeland components across the western U.S. using Landsat imagery from 1985-2021.
    The RCMAP product suite consists of nine fractional components: annual herbaceous, bare ground, herbaceous, litter, non-sagebrush shrub, perennial herbaceous, sagebrush, shrub, and tree, in addition to the temporal trends of each component. Several enhancements were made to the RCMAP process relative to prior generations. First, we have trained time-series predictions directly from 331 high-resolution sites collected from 2013-2018 from Assessment, Inventory, and Monitoring (AIM) instead of using the 2016 “base” map as an intermediary. This removes one level of model error and allows the direct association of high-resolution derived training data to the corresponding year of Landsat imagery. We have incorporated all available (as of 10/1/22) Bureau of Land Management (BLM), Assessment, Inventory, and Monitoring (AIM), and Landscape Monitoring Framework (LMF) observations. LANDFIRE public reference database training observations spanning 1985-2015 have been added. Neural network models with Keras tuner optimization have replaced Cubist models as our classifier. We have added a tree canopy cover component. Our study area has expanded to include all of California, Oregon, and Washington; in prior generations landscapes to the west of the Cascades were excluded. Additional spectral indices have been added as predictor variables, tasseled cap wetness, brightness, and greenness. Location information (i.e., latitude and longitude/ x and y coordinates) and elevation above sea level have been added as predictor variables. CCDC-Synthetic Landsat images were obtained for 6 monthly periods for each region and were added as predictors. These data augment the phenologic detail of the 2 seasonal Landsat composites.
    Post-processing has been improved with updated fire recovery equations stratified by ecosystem resistance and resilience (R and R) classes (Maestas and Campbell 2016) to stratify recovery rates. Ecosystem R and R maps are only available for the sagebrush biome. We intersected classes with 1985-2020 average water year precipitation to identify precipitation thresholds corresponding to R and R classes. Outside of the sagebrush biome, precipitation was used to produce R and R equivalent (low, medium, high). Due to the fast recovery following fire in California chapparal (e.g., Keeley and Keeley 1981, Storey et al. 2016), we used EPA level 3 ecoregions to define a 4th R and R zone. Recovery rates are based on (Arkle et al (in press)) who evaluated the recovery of plant functional groups in 1278 post-fire rehab plots by time since disturbance stratified by ecosystem resistance and resilience. We have expanded this analysis by evaluated postfire-recovery in all AIM and LMF data across the West to establish maximum sage, shrub, and tree cover by time-since fire. Recovery limits in California follow (Keeley and Keeley 1981 and Storey et al. 2016). Second, post-processing has been enhanced through a revised noise detection model. For each pixel, we fit a third order polynomial model for each component cover time-series. Observations with a z-score more than 2 standard deviations from the mean are removed, and a new third order polynomial model (i.e., cleaned fit) is fit to observations within this threshold. Finally, looking again at all observations, those observations with a z-score more than 2 standard deviations from the mean of the cleaned fit are replaced with the mean of the prior and subsequent year component cover values.
    Processing efficiency has been increased using open-source software and USGS High-Performance Computing (HPC) resources. The mapping area included eight regions which were subsequently mosaicked for all nine components. These data can be used to answer critical questions regarding the influence of climate change and the suitability of management practices.
    Component products can be downloaded https://www.mrlc.gov/data.

~~~~~
Update **Citation** to: 

Rigge, M.B., Bunde, B., Postma, K., Shi, H., 2022, Rangeland Condition Monitoring Assessment and Projection (RCMAP) Fractional Component Time-Series Across the Western U.S. 1985-2021: U.S. Geological Survey data release, https://doi.org/10.5066/P9ODAZHC.
    [doi:10.5066/P9ODAZHC](https://doi.org/10.5066/P9ODAZHC)

Add Tree **Band**: 

    {
        name: 'rangeland_tree',
        description: |||
          Percent of the pixel covered by tree.
        |||,
        'gee:units': '%',
      },
      {

   },
    rangeland_tree: {
      minimum: 0.0,
      maximum: 100.0,
      'gee:estimated_range': false,
    },

schwehr commented:
Closing this PR:

- Some of the changes here are in main
- The templating that added in f03b8038b358f1572f5c9ec410f5d30fb6315e1f after this CL was created a conflict
- a4c7db14019c744d9c95118b50b0ec93523db427 created [USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_COVER.jsonnet](https://github.com/google/earthengine-catalog/blob/a4c7db14019c744d9c95118b50b0ec93523db427/catalog/USGS/NLCD_RELEASES/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_COVER.jsonnet) 
- Rather than resolving with a `git rebase`, I'll undo the templating and make sure that all the changes are in main.  Once we've made sure all the existing desired changes are in, we can move forward from a fresh branch from the head of the main branch.
- Using the main branch for pull requests can complicate things, so we'll see if we can get a workflow that does separate branches

Note the title of this PR refers to `COVER`, but the dataset being described is `USGS/NLCD_RELEASES/USGS_NLCD_RELEASES_2019_REL_RCMAP_V5_COVER`
